ffi/lru.lua

```diff
@@ -74,34 +74,71 @@ function lru.new(max_size, max_bytes)
         end
     end
 
-    local function del(key, tuple)
-        tuple[VALUE]:onFree() -- KOReader
-        map[key] = nil
-        cut(tuple)
-        size = size - 1
-        bytes_used = bytes_used - (tuple[BYTES] or 0)
-        removed_tuple = tuple
+    local del
+    if enable_eviction_cb then
+        if max_bytes then
+            del = function(key, tuple)
+                tuple[VALUE]:onFree()
+                map[key] = nil
```

Comment:
No way to avoid `enable_eviction_cb` and having to know beforehand what kind of stuff we'll store (stuff with :onFree, or without) ?
Can't we store the presence or not of `item.onfree` at insertion time in `tuple[HAS_ONFREE]` ? (`local HAS_ONFREE = 6`)

Reply:
Not really, not without passing something at instantiation time. And delaying a check to insert time defeats the purpose of the whole endeavor ;).